core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -162,10 +178,14 @@ private ManifestFile copyManifest(ManifestFile manifest) {
 
     validateFilesCounts();
 
+    // TODO: add sequence numbers here
+    Iterable<ManifestFile> newManifestsWithMetadata = Iterables.transform(
```

Comment:
This place requires attention. When we introduce sequence numbers, we will have to iterate through all new manifests. For now, iterating through `newManifests` and `rewrittenAddedManifests` is redundant. However, I expect we will add sequence number quickly and we will simply need to change the closure.